src/libs/defaultRedirection/defaultRedirection.ts

```diff
@@ -0,0 +1,131 @@
+  get: (path: string) => Promise<InstanceSettings>
+}
+
+interface InstanceSettings {
+  data: {
+    attributes: {
+      default_redirection: string
+    }
+  }
+}
+
+export const isDefaultRedirectionUrlNavigationApp = (
+  defaultRedirectionUrl: string,
+  client: CozyClient
+): boolean => {
+  const subdomainType = client.capabilities.flat_subdomains ? 'flat' : 'nested'
+
+  const { slug } = deconstructCozyWebLinkWithSlug(
+    defaultRedirectionUrl,
+    subdomainType
+  )
+
+  return slug === NAVIGATION_APP_SLUG
+}
+
+export const fetchDefaultRedirectionUrl = async (
+  client: CozyClient
+): Promise<string | undefined> => {
+  let defaultRedirectionUrl
+
+  try {
+    const settingsCollection = client.collection(
+      'io.cozy.settings'
+    ) as SettingsCollection
+    const { data } = await settingsCollection.get('instance')
+
+    const defaultRedirection = data.attributes.default_redirection
+
+    defaultRedirectionUrl = formatRedirectLink(defaultRedirection, client)
+  } catch {
+    defaultRedirectionUrl = undefined
+  }
+
+  return defaultRedirectionUrl
+}
+
+export const setDefaultRedirectionUrl = async (
+  defaultRedirectionUrl: string
+): Promise<void> => {
+  await AsyncStorage.setItem(
+    strings.DEFAULT_REDIRECTION_URL_STORAGE_KEY,
+    defaultRedirectionUrl
+  )
+}
+
+export const getDefaultRedirectionUrl = async (): Promise<string | null> => {
+  return await AsyncStorage.getItem(strings.DEFAULT_REDIRECTION_URL_STORAGE_KEY)
+}
+
+export const fetchAndSetDefaultRedirectionUrl = async (
+  client: CozyClient
+): Promise<void> => {
+  const newDefaultRedirectionUrl = await fetchDefaultRedirectionUrl(client)
+
+  if (!newDefaultRedirectionUrl) return
+
+  await setDefaultRedirectionUrl(newDefaultRedirectionUrl)
+}
+
+export const fetchAndSetDefaultRedirectionUrlInBackground = (
+  client: CozyClient,
+  delayInMs = DEFAULT_REDIRECTION_DELAY_IN_MS
+): Promise<void> => {
+  return new Promise(resolve => {
+    setTimeout(() => {
+      fetchAndSetDefaultRedirectionUrl(client)
+        .then(resolve)
+        .catch(err =>
+          log.error(
+            `Something went wrong when fetching and setting default redirection url in background: ${getErrorMessage(
+              err
+            )}`,
+            resolve()
+          )
+        )
+    }, delayInMs)
+  })
+}
+
+export const getOrFetchDefaultRedirectionUrl = async (
+  client: CozyClient
+): Promise<string | undefined> => {
+  const localDefaultRedirectionUrl = await getDefaultRedirectionUrl()
+  if (localDefaultRedirectionUrl) {
+    void fetchAndSetDefaultRedirectionUrlInBackground(client)
+
+    if (
+      isDefaultRedirectionUrlNavigationApp(localDefaultRedirectionUrl, client)
+    )
+      return
+
+    return localDefaultRedirectionUrl
+  } else {
+    const newDefaultRedirectionUrl = await fetchDefaultRedirectionUrl(client)
+
+    if (!newDefaultRedirectionUrl) return
+
+    await setDefaultRedirectionUrl(newDefaultRedirectionUrl)
```

Comment:
Here you can reuse `fetchAndSetDefaultRedirectionUrl()` (if you add a return to it)

Before
```ts
const newDefaultRedirectionUrl = await fetchDefaultRedirectionUrl(client)

if (!newDefaultRedirectionUrl) return

await setDefaultRedirectionUrl(newDefaultRedirectionUrl)
```

after
```ts
const newDefaultRedirectionUrl = await fetchAndSetDefaultRedirectionUrl(client)
```

Reply:
And after that, the code would be pretty similar between the `if` and the `else`, so you can condition only the `fetchAndSetDefaultRedirectionUrlInBackground` vs `fetchDefaultRedirectionUrl` and do the `if (isDefault...)` and the `return` outside of the if